core/state.go

```diff
@@ -18,6 +19,25 @@ func MakeOperatorSocket(nodeIP, dispersalPort, retrievalPort string) OperatorSoc
 	return OperatorSocket(fmt.Sprintf("%s:%s;%s", nodeIP, dispersalPort, retrievalPort))
 }
 
+func ParseOperatorSocket(operatorSocket string) (host string, dispersalPort string, retrievalPort string, err error) {
+	s := strings.Split(operatorSocket, ":")
+	if len(s) != 2 {
+		err = fmt.Errorf("invalid socket address format: %s", operatorSocket)
+		return
+	}
+	host = s[0]
+	dispersalPort = s[1]
```

Comment:
Is this correct? If "host:123;456" is split by ":", then the s[1] will be "123;456", right?

Reply:
Nice catch!